packages/flutter/lib/src/rendering/sliver_grid.dart

```diff
@@ -105,17 +126,77 @@ class SliverGridGeometry {
 abstract class SliverGridLayout {
   /// Abstract const constructor. This constructor enables subclasses to provide
   /// const constructors so that they can be used in const expressions.
-  const SliverGridLayout();
+  const SliverGridLayout({ this.layoutType = SliverGridLayoutType.fixed });
 
-  /// The minimum child index that intersects with (or is after) this scroll offset.
-  int getMinChildIndexForScrollOffset(double scrollOffset);
+  /// TODO(Piinks): Add docs
+  final SliverGridLayoutType layoutType;
+
+  /// The minimum child index that intersects with (or is after) this scroll
+  /// offset.
+  /// TODO(Piinks): Add more docs - only used for fixed layouts
+  int getMinChildIndexForScrollOffset(double scrollOffset) {
+    return 0;
+    // assert(
```

Comment:
I've removed several assertions here. I don't know that they are necessary, we can revisit them when we are getting closer to final designs.